airflow/serialization/serialized_objects.py

```diff
@@ -350,6 +373,8 @@ class SerializedBaseOperator(BaseOperator, BaseSerialization):
         if v.default is not v.empty
     }
 
+    dependency_detector = DependencyDetector()
```

Comment:
Why do we need a new instance of DependencyDetector?

Reply:
not necessarily